components/CompanyRegister.tsx

```diff
@@ -1,42 +1,72 @@
-import { CreateCompanyItem } from '@/db/entities/company';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import { useForm } from 'react-hook-form';
+import { useAccount } from 'wagmi';
 import useCompany from '@/hooks/useCompanyRegistration';
-import { isSuccessResponse } from '@/pages/api/companies';
+import { useRecruitmentRegisterCompany } from '../src/generated';
 import { useRouter } from 'next/router';
 import { useEffect, useState } from 'react';
 import { toast } from 'react-toastify';
-import { useAccount } from 'wagmi';
-import { useRecruitmentRegisterCompany } from '../src/generated';
+import { isSuccessResponse } from '@/pages/api/companies';
 import { Badge } from './ui/badge';
 
+const schema = yup.object().shape({
+  companyName: yup.string().required('Company Name is required'),
+  companyEmail: yup
+    .string()
+    .email('Invalid email')
+    .required('Email is required'),
+  companySite: yup
+    .string()
+    .matches(
+      /^(https?:\/\/)?(www\.)?([A-Za-z0-9-._~:/?#[\]@!$&'()*+,;=]|%[A-Fa-f0-9]{2})*$/,
+      'Invalid website URL'
+    )
+  });
+
+interface FormData {
+  companyName: string;
+  companyEmail: string;
+  companySite: string;
+}
+
 const CompanyRegister = () => {
   const router = useRouter();
   const { address } = useAccount();
-  const { isValidating, isRegistered } = useCompany(address);
```

Comment:
form submit button should be disabled until `isValidating`=== false.

So this needs to be added back in

Reply:
re-added

Reply:
`isValidating` is from the `useCompany` hook.  The hook checks if the company is already registered.  If the company is registered then the register button should be disabled. `isValidating` is true while the hook is checking if the company is registered.  We want to also disable the register button while checking if the company is registered.